server/src/main/java/org/opensearch/common/blobstore/transfer/RemoteTransferContainer.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.blobstore.transfer;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.common.SetOnce;
+import org.opensearch.common.Stream;
+import org.opensearch.common.StreamProvider;
+import org.opensearch.common.TransferPartStreamSupplier;
+import org.opensearch.common.blobstore.stream.StreamContext;
+import org.opensearch.common.blobstore.stream.write.WriteContext;
+import org.opensearch.common.blobstore.stream.write.WritePriority;
+import org.opensearch.common.blobstore.transfer.stream.OffsetRangeInputStream;
+import org.opensearch.common.blobstore.transfer.stream.ResettableCheckedInputStream;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Path;
+import java.util.Objects;
+
+/**
+ * RemoteTransferContainer is an encapsulation for managing transfers for translog and segment files.
+ */
+public class RemoteTransferContainer implements Closeable {
+
+    private int numberOfParts;
+    private long partSize;
+    private long lastPartSize;
+
+    private final long contentLength;
+    private final SetOnce<ResettableCheckedInputStream[]> inputStreams = new SetOnce<>();
+    private final String localFileName;
```

Comment:
it's possible that local file might not exist? For eg in case of metadata files, we would not have the file existing on local disk. For translog metadata file, we are not creating any file but directly creating the stream and uploading the blob.

Reply:
This is a good point and I think with new change where caller is providing a stream supplier, we can look into removing local file references.

Reply:
@raghuvanshraj